plugins/woo-ai/src/image-background-removal/background-removal-link.tsx

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+import { __experimentalUseBackgroundRemoval as useBackgroundRemoval } from '@woocommerce/ai';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import MagicIcon from '../../assets/images/icons/magic.svg';
+import { FILENAME_APPEND } from './constants';
+import {
+	uploadImageToLibrary,
+	getCurrentAttachmentDetails,
+} from './image_utils';
+
+const getErrorMessage = ( errorCode?: string ) => {
+	switch ( errorCode ) {
+		case 'invalid_image_file':
+			return __( 'Invalid image', 'woocommerce' );
+		case 'image_file_too_small':
+			return __( 'Image too small', 'woocommerce' );
+		case 'image_file_too_large':
+			return __( 'Image too large', 'woocommerce' );
+		default:
+			return __( 'Something went wrong', 'woocommerce' );
+	}
+};
+
+export const BackgroundRemovalLink = () => {
+	const { fetchImage } = useBackgroundRemoval();
+	const [ state, setState ] = useState< '' | 'generating' | 'uploading' >(
```

Comment:
Let's use an enum here, like the other components.

Reply:
This serves more-or-less the same purpose as an explicit enum in TypeScript. There are some [pros and cons](https://www.bam.tech/en/article/should-you-use-enums-or-union-types-in-typescript), but for a simple use case I tend to prefer just using a union like this.

What do you think?

Reply:
Thanks for including the link. I prefer an enum since you get autocomplete, and there's less chance of a typo messing things up, especially since we're setting words as states, not numbers. But it's totally up to you.

Reply:
You actually should still get autocomplete (I do in vscode), and at least according to the article above they should be more immediate than an enum. Typescript will also let you know about any typos or anything with the union approach as well:
![image](https://github.com/woocommerce/woocommerce/assets/444632/9d35ab10-7929-45f2-a031-e37f14e87cb6)